Fix/BAC-171--176: Employee controller endpoints

ruxy1212 opened the pull request:
#### Task reference
BAC-171
BAC-172
BAC-173
BAC-174
BAC-175
BAC-176
#### Summary
Tested all endpoints under the EmployeeController and fixed all issues.
- Fixed JWT authentication error: changed `$token = explode(" ", $header)[1];` to `$token = explode(" ", $header)[0];`
- Fixed responses, using proper HTTP response codes
- Fixed error on this line `!$orgData->count()` which performs may perform an operation on null data, changed to `is_null($employee)`
- Changed db::insert to eloquent ::create(), to ensure automatic creation of created_at and updated_at columns.

#### References 
![new10](https://user-images.githubusercontent.com/85977511/206630745-e51ba64d-29bd-4f82-a3bb-3c0ee13da11d.PNG)
1. Fixed updated_at and created_at auto-insert

![new9](https://user-images.githubusercontent.com/85977511/206630749-9448025b-5689-44ae-9795-74795b85c75d.PNG)
2. Get all employees, with total count

![new8](https://user-images.githubusercontent.com/85977511/206630750-a91dfc24-d8b4-4abb-b58d-c87379b563d7.PNG)
3. Updated employee by id 

![new7](https://user-images.githubusercontent.com/85977511/206630754-1e6e59e5-d07c-4846-aca1-6dc610e04701.PNG)
4. Get all employees under a particular organization by id with count

![new6](https://user-images.githubusercontent.com/85977511/206630756-69b5b5f2-71fc-4c7c-8227-ad002b8aed1c.PNG)

![new5](https://user-images.githubusercontent.com/85977511/206630758-3f6e5d4f-f6c6-445e-92d5-d8d670f2b2a4.PNG)
5. Fetch details of a particular employee by id

![new4](https://user-images.githubusercontent.com/85977511/206630764-d1d0c7c3-0307-4b76-a440-b12c1f6249a9.PNG)
6. Add new employee manually

![new3](https://user-images.githubusercontent.com/85977511/206630766-34eadff0-206a-4ad4-aa90-c8e27533a350.PNG)
7. Add new employee manually, email already exist response

![new2](https://user-images.githubusercontent.com/85977511/206630769-4d1648c6-3ccb-4313-85fd-7ce68ee3443b.PNG)
8. Confirm employees to add by csv

![new1](https://user-images.githubusercontent.com/85977511/206630771-c8a60753-7c0b-4ee5-9fb7-844352195d56.PNG)
9. Add employee by csv, parsing csv and returning employee data with org_id and department_id columns

ruxy1212 committed:
Fix(BAC-171-176): Employee controller endpoints refactor and testing plus invalid Jwt token error fix

ruxy1212 committed:
Fix(BAC-171-176): Employee controller endpoints refactor and testing plus invalid Jwt token error fix

ruxy1212 committed:
Merge branch 'dev' of https://github.com/workshopapps/engineerprofile360.web into fix/BAC-171--176-employee-controller-endpoints

ruxy1212 committed:
Fix: Authorization header, changed back to index[1], csv_parser to remove empty arrays while parsing

dasaint001 commented on backend/app/Http/Controllers/EmployeeController.php:
```diff
@@ -220,25 +221,25 @@ public function getEmplyeesByDepartment($departmentId)
             );
         } catch (\Exception $e) {
-            return $this->sendResponse(true, 'Employees not fetched', $e->getMessage(), null,  Response::HTTP_INTERNAL_SERVER_ERROR);
+            return $this->sendResponse(true, $e->getMessage(), 'Employees not fetched', null,  Response::HTTP_INTERNAL_SERVER_ERROR);
```
what happened here?

dasaint001 commented on backend/app/Http/Controllers/EmployeeController.php:
```diff
@@ -220,25 +221,25 @@ public function getEmplyeesByDepartment($departmentId)
                 null,
-                'All employees',
-                $companies,
+                "All employees (count: $totalCount)",
```
why are you returning just count here? This belongs to admin... You are not expected to change this.. There's another endpoint doing this

ruxy1212 replied:
That's just a response abi? I only made it more descriptive.
All employees >>>> All employees (count: 3)

No wahala sha. I don change am

Then about the response, this is the format, at least that was what I got from Ben's implementation on authentication controller.
`sendResponse(true, $e->getMessage(), "Something went wrong logging in, please try again", null, 500);`
``` bool $errorState = true,
        string $error = null,
        string $message = 'OK',
        $data = null,
        int $code = 200
```
I don't know who decided to interchange some of the responses under my controller, maybe I did so myself but decided to use a uniform response for all. For my try blocks, $error is the exception error [$e->getMessage()], and $message is my own description of the error.

the $message should be the error while the $error should be passed to the message not the way you are doing it.

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

dasaint001 commented on backend/app/Http/Controllers/EmployeeController.php:
```diff
@@ -291,7 +291,7 @@ public function deleteEmployee($employeeId): JsonResponse
             return $this->sendResponse(false, null, 'Employee deleted successfully', null, Response::HTTP_OK);
         } catch (Exception $e) {
-            return $this->sendResponse(true, "Could not fetch employee ", $e->getMessage(), null,  Response::HTTP_INTERNAL_SERVER_ERROR);
+            return $this->sendResponse(true, $e->getMessage(), "Could not fetch employee ", null,  Response::HTTP_INTERNAL_SERVER_ERROR);
```
this is not written by you... so you can't change it.

ruxy1212 replied:
So I should leave some with message before error and some error before message? Because I didn't write it and yet I was assigned to manage it. Wow.

No wahala. I think say na my controller make I sanitize am. Obviously its yours. No issues

dasaint001 replied:
lol... no ownership here. we all did this together... I am not the owner of anything...

dasaint001 reviewed (changes requested):
I don't think what you are doing here is necessary.. pls don't touch what was not written by you. if there's any bug, pls fix... And issue of sendresponse is not a bug. error comes before message.. not message before error. Pls you have to fix the bugs if any and if none, pls leave it as it is.. Thanks

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

ruxy1212 committed:
Made review changes to fix/BAC-171--176-employee-controller-endpoints

ruxy1212 commented:
I've done all the changes boss. 

- The only ones remaining are inserting before checking if organization exists, It caused some issues at the frontend when trying to retrieve an invalid org_id.
- Then the other one I changed to `is_null($orgData)`, this is because performing a count on a find() that returned null will crash, so it's better to check if the returned data is null instead.
- I also changed the `db::insert` to create() as I explained in my PR description up that the insert query builder makes `created_at` and `updated_at` columns to be NULL

ruxy1212 commented:
Any update yet sir?

ruxy1212 commented:
Thanks boss
